target/linux/meson/image/Makefile

```diff
@@ -0,0 +1,80 @@
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+# makefile mod by lunatickochiya ain at s805 soc  mostly onecloud
+#
+include $(TOPDIR)/rules.mk
+include $(INCLUDE_DIR)/image.mk
+
+FAT32_BLOCK_SIZE=1024
+FAT32_BLOCKS=$(shell echo $$(($(CONFIG_TARGET_KERNEL_PARTSIZE)*1024*1024/$(FAT32_BLOCK_SIZE))))
+
+define Build/boot-script
+	# Make an U-boot image and copy it to the boot partition
+	mkimage -A arm -O linux -T script -C none -a 0 -e 0 -n "boot.scr" -d boot.txt  $(KDIR)/boot.scr
+endef
+
+define Build/sdcard-common
+	$(RM) -f $@.boot
+	mkfs.fat -C $@.boot $(FAT32_BLOCKS)
+
+	mkdir -p $(KDIR)/boot.fat
+
+	$(CP) $(KDIR)/boot.scr $(KDIR)/boot.fat/boot.scr
+	mcopy -i $@.boot $(KDIR)/boot.scr ::
+	$(CP) $(IMAGE_KERNEL) $(KDIR)/boot.fat/uImage
+	mcopy -i $@.boot $(KDIR)/boot.fat/uImage ::
+
+	$(foreach dts,$(shell echo $(DEVICE_DTS)),$(CP) $(DTS_DIR)/$(dts).dtb $(KDIR)/boot.fat/dtb;)
+	mcopy -i $@.boot $(KDIR)/boot.fat/dtb ::
+
+	$(RM) -rf $(KDIR)/boot.fat
+
+	./gen_aml_sdcard_img.sh $@ $@.boot $(IMAGE_ROOTFS) \
+		$(CONFIG_TARGET_KERNEL_PARTSIZE) $(CONFIG_TARGET_ROOTFS_PARTSIZE)
+endef
+
+### Devices ###
+define Device/Default
+  FILESYSTEMS := ext4
+  IMAGES := sdcard.img
+  KERNEL_DEPENDS = $$(wildcard $(DTS_DIR)/$$(DEVICE_DTS).dts)
+  KERNEL_LOADADDR := 0x01080000
+  KERNEL_NAME := Image
+  KERNEL := kernel-bin | uImage none
+  PROFILES = Default $$(DEVICE_NAME)
+endef
+
+define Device/odroid-c1
+  DEVICE_DTS := meson8b-odroidc1
+  DEVICE_PACKAGES +=
+  DEVICE_TITLE := ODROID C1
+  KERNEL_LOADADDR := 0x00208000
+  IMAGE/sdcard.img := boot-script | sdcard-common $$(DEVICE_NAME)
+endef
+ifeq ($(SUBTARGET),meson8b)
```

Comment:
what? see how we did in rockchip. move these boards into a dedicated file, then include them via `include $(SUBTARGET).mk`